src/engine-actions/close-channels.js

```diff
@@ -0,0 +1,83 @@
+const {
+  listChannels,
+  closeChannel,
+  listPendingChannels
+} = require('../lnd-actions')
+/**
+ * Closes a channel given the channel point
+ *
+ * @param {String} paymentChannelNetworkAddress
+ * @param {String} fundingAmount - int64 string
+ * @returns {Promise<void>} returns void on success
+ */
+async function closeChannels ({ force = false, pending = false } = {}) {
+  const { channels = [] } = await listChannels({ client: this.client })
+  const activeChannels = channels.filter(c => c.active)
+
+  let inactiveChannels = []
+  if (force === true) {
+    inactiveChannels = channels.filter(c => c.active === false)
+  }
+
+  let pendingChannels = []
+  if (pending === true) {
+    const { pendingOpenChannels = [] } = await listPendingChannels({ client: this.client })
+    pendingChannels = pendingOpenChannels
+  }
+  const closedChannelResponses = await Promise.all(activeChannels.concat(inactiveChannels).map(channel => close(channel, this.client, this.logger)))
+  const closedPendingChannelResponses = await Promise.all(pendingChannels.map(channel => close(channel.channel, this.client, this.logger)))
```

Comment:
For these 2 lines, instead of passing `this.channel` and `this.logger` we can use [`call`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/call).

Example:
```
const closedPendingChannelResponses = await Promise.all(pendingChannels.map(channel => close.call(this, channel.channel)))
```